### ISSUE TITLE
Can't create projects on Che

### ISSUE THREAD

**Issue by johnmcollier** [eclipse/codewind#1665]
**Codewind version:** latest

**Che version:** 7.6.0
**IDE extension version:** latest
**Kubernetes cluster:** OCP 4.2

**Description:**
Starting today, I've been unable to create projects on Che, as the dialog doesn't show up when I click the "+" button or use the "Create project" command in the command pallet. 

I see this in the logs each time I try to open the view:
```
root ERROR [hosted-plugin: 36] Promise rejection not handled in one second: TypeError: sources.map is not a function , reason: TypeError: sources.map is not a function
With stack trace: TypeError: sources.map is not a function
    at /tmp/vscode-unpacked/eclipse.codewind-theia.latest.gmlagxuqdv.codewind-theia.vsix/extension/out/command/connection/CreateUserProjectCmd.js:154:30
    at Generator.next (<anonymous>)
    at fulfilled (/tmp/vscode-unpacked/eclipse.codewind-theia.latest.gmlagxuqdv.codewind-theia.vsix/extension/out/command/connection/CreateUserProjectCmd.js:15:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

I'm unable to reproduce on a workspace on Che 7.6.0 from yesterday. The PFE logs don't show anything suspicious either.

**Steps to reproduce:**
1. Create a Che workspace from the [latest devfile](https://raw.githubusercontent.com/eclipse/codewind-che-plugin/master/devfiles/latest/devfile.yaml)
2. Wait til Theia connects to Codewind
3. Try to create a project. Nothing will happen in the UI.

**Comment by johnmcollier** [comment]
Marking as stopship as a core functionality is broken and will block SVT test efforts.

**Comment by johnmcollier** [comment]
Looks to be a cwctl issue? Looking at the output from codewind.log, I see the following from cwctl:
```
[INFO: 17:57:49.594 CLIWrapper.js:129]: Running CLI command: cwctl --json --insecure templates repos list --conid local
[DBUG: 17:57:49.627 CLIWrapper.js.on():192]: Successfully ran CLI command cwctl --json --insecure templates repos list --conid local, Output was:
null
[ERRO: 17:57:49.628 CLIWrapper.js.on():197]: Missing expected JSON output from CLI command, output was "null
"
[ERRO: 17:57:49.696 extension.js.on():44]: Unhandled promise rejection:
{}

```

It appears that `cwctl --json --insecure templates repos list --conid local` is returning null

**Comment by tetchel** [comment]
https://github.com/eclipse/codewind-vscode/pull/373 kind of works around the issue, and shows that error message a little more obviously.
once it's in. you will be able to create projects, the template source management page will still not work though.

**Comment by maysunfaisal** [comment]
Also cannot add image registry on 7.6.0 now. Not sure if it should be a separate issue or its because of the same reason as this issue

```
root INFO >>> Storing the layout...
root INFO <<< The layout has been successfully stored.
root ERROR [hosted-plugin: 31] [ERRO: 18:35:28.798 RegistriesPageWrapper.js:88]: Failed to add new image registry TypeError: templateSources.filter is not a function
    at TemplateSourcesList.<anonymous> (/tmp/vscode-unpacked/eclipse.codewind-theia.latest.gjudohvkhb.codewind-theia.vsix/extension/out/codewind/connection/TemplateSourceList.js:58:18)
    at Generator.next (<anonymous>)
    at /tmp/vscode-unpacked/eclipse.codewind-theia.latest.gjudohvkhb.codewind-theia.vsix/extension/out/codewind/connection/TemplateSourceList.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/tmp/vscode-unpacked/eclipse.codewind-theia.latest.gjudohvkhb.codewind-theia.vsix/extension/out/codewind/connection/TemplateSourceList.js:4:12)
    at TemplateSourcesList.hasCodewindSourceEnabled (/tmp/vscode-unpacked/eclipse.codewind-theia.latest.gjudohvkhb.codewind-theia.vsix/extension/out/codewind/connection/TemplateSourceList.js:55:16)
    at Object.<anonymous> (/tmp/vscode-unpacked/eclipse.codewind-theia.latest.gjudohvkhb.codewind-theia.vsix/extension/out/codewind/connection/RegistryUtils.js:118:97)
    at Generator.next (<anonymous>)
    at fulfilled (/tmp/vscode-unpacked/eclipse.codewind-theia.latest.gjudohvkhb.codewind-theia.vsix/extension/out/codewind/connection/RegistryUtils.js:15:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
root INFO [hosted-plugin: 31] [DBUG: 18:35:28.799 Requester.js:448]: Doing GET request to https://localhost:9090/api/v1/registrysecrets
root INFO [hosted-plugin: 31] [DBUG: 18:35:28.809 Requester.js:448]: Doing GET request to https://localhost:9090/api/v1/imagepushregistry
root INFO [hosted-plugin: 31] [DBUG: 18:35:28.816 Requester.js:139]: No image push registry is set
```

![Screen Shot 2020-01-08 at 1 34 38 PM](https://user-images.githubusercontent.com/31771087/72005652-2043aa80-321c-11ea-84d8-290bcce9b2a2.png)

**Comment by tetchel** [comment]
same issue

**Comment by malincoln** [comment]
@tobespc pls review

**Comment by markcor11** [comment]
reproduced with above dev file : 

![Screenshot 2020-01-09 at 14 05 24](https://user-images.githubusercontent.com/28102564/72074049-16d44400-32e9-11ea-9990-0cffe8e1c245.png)

**Comment by markcor11** [comment]
Looks like this may be related to recent changes in installer PR 303 - debugging....

**Comment by markcor11** [comment]
Have this working now by detecting if running on CHE before performing any Docker container checks. 

![Screenshot 2020-01-09 at 16 21 18](https://user-images.githubusercontent.com/28102564/72085485-4d1bbe80-32fd-11ea-9af7-307dc5df5dda.png)

**Comment by johnmcollier** [comment]
@markcor11 Slight nitpick, rather than performing a Docker container check, could we just check for the existence of one of the Che workspace env vars? Such as `CHE_WORKSPACE_ID`?

**Comment by markcor11** [comment]
@johnmcollier that sir is exactly how I solved it, swapping the host name search order from Docker container check first then a CHE check, to now doing the CHE check first by looking up the env var CHE_API_EXTERNAL.  If its set pfe host will be https:/localhost:9090 (the theia proxy) rather than attempt to find the docker containers.

**Comment by jagraj** [comment]
I got the latest drivers and I am still hitting this problem. I can not create image push registry from Image registry manager. 

<img width="528" alt="Screen Shot 2020-01-10 at 4 55 14 PM" src="https://user-images.githubusercontent.com/713849/72189636-58d5b680-33cb-11ea-9e0f-8c9a148bd531.png">

**Comment by DavidG1011** [comment]
Seeing the same as Jag on OCP 4.2 with Che 7.5.1:

![Screen Shot 2020-01-10 at 5 09 27 PM](https://user-images.githubusercontent.com/30476587/72189880-06e16080-33cc-11ea-8341-5df100c6cc5e.png)

Also still seeing this error when creating a new project:

![Screen Shot 2020-01-10 at 5 08 54 PM](https://user-images.githubusercontent.com/30476587/72189957-3db77680-33cc-11ea-96ea-7e2570d621c9.png)


[hosted-instance-log.txt](https://github.com/eclipse/codewind/files/4048327/hosted-instance-log.txt)

**Comment by markcor11** [comment]
I suspect you don’t have the latest CLI in the theia yet

**Comment by markcor11** [comment]
Just checked the pipeline status,  CWCTL containing the fix has still not yet been pushed as of Sunday 13:25 GMT.  Although build and test phases are complete, Jenkins seems stuck in the Deploy stage along with one other job since Friday.

See. Build 62 in https://ci.eclipse.org/codewind/job/Codewind/job/codewind-installer/job/master

**Comment by markcor11** [comment]
<img width="1201" alt="Screenshot 2020-01-12 at 13 42 05" src="https://user-images.githubusercontent.com/28102564/72219774-7a58ae80-3541-11ea-8c94-5f5c5c9145f5.png">
<img width="840" alt="Screenshot 2020-01-12 at 13 42 36" src="https://user-images.githubusercontent.com/28102564/72219775-7c227200-3541-11ea-84db-179787852fed.png">

**Comment by johnmcollier** [comment]
@markcor11 I tried kicking off a rebuild to see if it would fix the hanging

**Comment by johnmcollier** [comment]
Even with the 0.8.0 devfile (with newest images and plugins), I can still reproduce. @markcor11 FYI, are you able to reproduce on 0.8.0?

**Comment by markcor11** [comment]
Jenkins still shows that the 0.8.0 branch has not completed and stuck for 2+ days : 
![image](https://user-images.githubusercontent.com/28102564/72285062-48774300-363a-11ea-9076-72b76d09a8e0.png)

**Comment by tetchel** [comment]
A cwctl build went through at 14:18 our time, so this should be fixed in 0.8 now.

**Comment by johnmcollier** [comment]
Verified fixed on the 0.8.0 devfile.
